packages/query-core/src/query.ts

```diff
@@ -560,6 +564,14 @@ export class Query<
             ...state,
             isInvalidated: true,
           }
+        case 'pageFetched':
+          return {
+            ...state,
+            fetchMeta: {
+              ...state.fetchMeta,
+              fetchedPages: state.fetchMeta.fetchedPages + 1,
+            },
+          }
```

Comment:
I'm a little concerned about the triple equals comparison of `state.fetchMeta` here:

https://github.com/TanStack/query/blob/04945f38b3f2ff82a597083bf1a36724f973bc7e/packages/query-core/src/query.ts#L419

because every `pageFetched` action will create a "new" `fetchMeta`. Maybe it would be better, for the internal state, to keep it on a separate state variable ?